operator/apis/platform/v1alpha1/central_types.go

```diff
@@ -150,35 +150,50 @@ func (c *CentralComponentSpec) GetAdminPasswordGenerationDisabled() bool {
 
 // CentralDBEnabled returns a bool if CentralDBSpec is not nil
 func (c *CentralComponentSpec) CentralDBEnabled() bool {
-	if c == nil {
+	if c == nil || c.DB == nil {
 		return false
 	}
-	return c.DB != nil
+
+	return c.DB.Enabled == CentralDBEnabledTrue
 }
 
 // CentralDBSpec defines settings for the "central db" component.
 type CentralDBSpec struct {
+	//+kubebuilder:default=Default
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,order=1
+	Enabled CentralDBEnabled `json:"enabled"`
```

Comment:
It seems there are a few related settings:
1. whether the operator should be provisioning a postgres instance
2. whether central should be using rocksdb or postgres
3. if the latter:
   - connect to the provisioned instance, or
   - to some other postgres connection string

With what you propose we would control this with 2 knobs:
- `spec.central.db.enabled`
- `spec.central.db.connectionString`

It makes sense that the operator is not provisioning a postgres if a connection string is specified. However it feels awkward that one needs to look at `connectionString` to determine (1). The result is there are a couple of places in the codebase which need to look at both `central.Spec.Central.CentralDBEnabled()` and `central.Spec.Central.DB.IsExternal()`.

How about this as a single tri-state `spec.central.storage` field instead, with the following values (exact wording TBD):
- `Embedded`: use rocksdb, don't provision a postgres
- `DB`: provision a postgres and use that; note: currently in tech preview; requires empty/null `connectionString`
- `External`: don't provision a postgres, use some other postgres connection string; note: currently in tech preview; requires non-empty `connectionString`
- `Default`: currently same as `Builtin`, in 3.74, same as `DB`

Reply:
It's already controlled with 2 knobs, this PR doesn't change that:
* whether `spec.central.db` is present or not
* `spec.central.db.connectionString`

I think we should not add the value `Embedded` (or `Builtin`) because we will phase this out in v3.74, as discussed with @msugakov in Slack.

And separating `DB` and `External` into 2 states might save us from looking at both `central.Spec.Central.CentralDBEnabled()` and `central.Spec.Central.DB.IsExternal()` in some places, but it also creates two possible invalid configurations.

So I'm still leaning towards how Johannes implemented it here.

Reply:
I agree, with Vlad on this. With the suggested change we would enable a user setting the type explicitly to embedded, which we intend to phase out by the next release.